src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -512,9 +499,10 @@ private void ExecuteClusterTransaction()
                     {
                         var batchSize = Configuration.Cluster.MaxClusterTransactionsBatchSize;
                         var executed = ExecuteClusterTransaction(context, batchSize);
-                        if (executed.Count == batchSize)
+                        if (executed.Count != 0)
```

Comment:
We changed the implementation.
A cluster-wide transaction without a database command will not be written to the `ClusterStateMachine.TransactionCommands` table.
Instead, if we don't find any new commands we notify according to the last compare exchange index